### PR TITLE
fix Makefile for Darwin

### PR DESCRIPTION
**Pull request by mhagmajer**
This fixes issue https://github.com/tass-belgium/picotcp/issues/461

### CODE REVIEW

**Comment by mhagmajer** [comment]
Shoutout to @basilfx for contributing on this!

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,9 @@
 -include ../../config.mk
 -include ../../tools/kconfig/.config
 
-CC:=$(CROSS_COMPILE)gcc
+OS:=$(shell uname)
+
+CC:=$(CROSS_COMPILE)clang
```
Comment (frederikvs) [inline Makefile]: I'd prefer to keep the default on GCC, otherwise the build would break for people who don't have clang installed.

File: modules/pico_dev_ppp.c [hunk 2]
```diff
@@ -302,6 +302,10 @@ static void lcp_optflags_print(struct pico_device_ppp *ppp, uint8_t *opts, uint3
 #define PPP_TIMER_ON_AUTH       0x10u
 #define PPP_TIMER_ON_IPCP       0x20u
 
+static int should_escape(uint8_t byte) {
+  return (byte == PPPF_FLAG_SEQ || byte == PPPF_CTRL_ESC || byte < 0x20);
+}
+
```
Comment (frederikvs) [inline modules/pico_dev_ppp.c]: This PPP stuff doesn't really belong in this PR I think, but I must admit that it's much nicer code :-)
But unless I'm mistaken, the `byte < 0x20` part isn't equivalent with the old code - what's your reasoning behind this change? (Not a huge expert on PPP myself)

File: Makefile (hunk 1 above)
Comment (mhagmajer) [inline Makefile]: Sure. In my project I compile picotcp for Android under x86 and arm which come with `clang` compiler rather than `gcc`. What would be the preferred way off supplying the compiler to use?

File: modules/pico_dev_ppp.c (hunk 2 above)
Comment (mhagmajer) [inline modules/pico_dev_ppp.c]: @frederikvs I tested it with a fiscal device that I have that operates on TCP/IP/PPP stack.

I think this is sort of escaping is implied by https://tools.ietf.org/html/rfc1662#section-4.2.

**Comment by mhagmajer** [comment]
Thanks for your comments @frederikvs! Please take a look at my replies while I'm looking into the CLA.

File: Makefile (hunk 1 above)
Comment (frederikvs) [inline Makefile]: You could run `make CC=clang`, that overrides the default that's set in the file

File: modules/pico_dev_ppp.c (hunk 2 above)
Comment (frederikvs) [inline modules/pico_dev_ppp.c]: @thijsdv probably knows more about PPP than I do.
Thijs, could you have a look whether this modification makes sense to you as well?

Comment (thijsdv) [inline modules/pico_dev_ppp.c]: I'll check it out this weekend (hopefully)

Comment (thijsdv) [inline modules/pico_dev_ppp.c]: Seems good to me, but the RFC states it should also escape "0x03" (= PPPF_CTRL in the module), I would add that one for completeness

File: modules/pico_dev_ppp.c [hunk 3]
```diff
@@ -341,7 +345,7 @@ static int ppp_serial_send_escape(struct pico_device_ppp *ppp, void *buf, int le
     /* Start byte. */
     out_buf[0] = in_buf[0];
     for(i = 1, j = 1; i < (len - 1); i++) {
-        if (((in_buf[i] + 1u) >> 1) == 0x3Fu) {
+        if (should_escape(in_buf[i])) {
```
Comment (thijsdv) [inline modules/pico_dev_ppp.c]: This commit is breaking the unit tests. Did you forgot to shift it by one or could this be a mis-interpretation of you @danielinux ?

File: modules/pico_dev_ppp.c (hunk 2 above)
Comment (mhagmajer) [inline modules/pico_dev_ppp.c]: @thijsdv cool, let me add that too! thanks

Comment (mhagmajer) [inline modules/pico_dev_ppp.c]: Created a new pull request with just this part: https://github.com/tass-belgium/picotcp/pull/463

File: Makefile (hunk 1 above)
Comment (mhagmajer) [inline Makefile]: Created a new pull request with just this part: https://github.com/tass-belgium/picotcp/pull/463
